internal/rename/agent/executor.go

```diff
@@ -0,0 +1,65 @@
+/*
+ *  *******************************************************************************
+ *  * Copyright (c) 2019 Edgeworx, Inc.
+ *  *
+ *  * This program and the accompanying materials are made available under the
+ *  * terms of the Eclipse Public License v. 2.0 which is available at
+ *  * http://www.eclipse.org/legal/epl-2.0
+ *  *
+ *  * SPDX-License-Identifier: EPL-2.0
+ *  *******************************************************************************
+ *
+ */
+
+package agent
+
+import (
+	"fmt"
+
+	"github.com/eclipse-iofog/iofog-go-sdk/pkg/client"
+	"github.com/eclipse-iofog/iofogctl/internal"
+
+	"github.com/eclipse-iofog/iofogctl/internal/config"
+	"github.com/eclipse-iofog/iofogctl/pkg/util"
+)
+
+func Execute( namespace, name, newname string) error {
+	// Get config
+	agent, err := config.GetAgent(exe.namespace, exe.name)
+	if err != nil {
+		return err
+	}
+
+	util.SpinStart(fmt.Sprintf("Renaming Agent %s", name))
+
+	// Init remote resources
+	clt, err := internal.NewControllerClient(exe.namespace)
+	if err != nil {
+		return err
+	}
+
+	ns, err := config.getNamespace(namespace)
+	if err != nil {
+		return err
+	}
+	for i, ag := range ns.Agents {
+		if ag.Name == name {
+			ns.Agents[i].name = newname
+			config.FlushConfig()
```

Comment:
This should be `config.Flush()` not FlushConfig. Flush flushed the content of namespaces. FlushConfig flushes the content of `~/.iofog/config.yaml`

Reply:
config.Flush should be defered, so that it is only done if the rest is successful